### PR TITLE
Fix/bunch o issues

### LINKED ISSUE #2355: Multiblock needs localization.
multiblock

### LINKED ISSUE #2391: broken permissions
### Minecolonies version
1.12.2-0.8.8117
version X.X.XXXX or curse X.X.XX

### Expected behavior
to work correctly
- XXX

### Actual behaviour
doesnt work rigth
- YYY

### Steps to reproduce the problem
set any fake player in permissions in the townhall to officer log out then back in and it is reset
1. XYZ
2. ZYX
3. ...
4. Profit

### LINKED ISSUE #2414: Some Colonists' Inventories Won't Open
### MineColonies Version

- 1.12.2-0.8.8147

### Forge Version

- 14.23.3.2674

### Expected Behavior

The colonist's inventory should appear when I click on "Inventory" in their menu.

### Actual Behavior

The colonist's inventory does not appear when I click on "Inventory" in their menu.
This is not the case for all colonists, however; some work just fine. After closing and reloading the game, it seems that the individual colonists this bug affects _may_ change, i.e. their inventory may open during one play session and not the next. I could be wrong.

### Steps to Reproduce the Problem

1. Load the game with MineColonies 1.12.2-0.8.8147.
2. Find a colonist.
3. Try to open the selected colonist's inventory.
4. If the inventory opens as usual, repeat steps 2 and 3 with another colonist. If nothing happens when you click "Inventory", continue to step 5.
5. You have encountered the problem.

### Note

I really don't know of anything else that would be helpful to solving this issue because I don't know what would be causing the issue. I know I was told to reopen the issue with more information, but there really isn't much else I can give you. I would include a video clip of me trying to open colonists' inventories, but I can't seem to get that working at the moment.

All I can say is that sometimes, the colonists' inventories won't open when I click the "Inventory" button, and that didn't start happening until I installed MineColonies 0.8.8147. I'll revert back to a previous version for now, and if I somehow encounter the issue again, I'll let you know.

Also, if any other players are also encountering this problem, please comment or upvote this issue!

### LINKED ISSUE #2415: [Feature Request] Add Town name to be displayed when barbarians attack.
A much needed feature for multiplayer settings.

When player receives a msg that barbarians are incoming, it should include the name of the Town that is being/going to be attacked by barbarians.

When players on server (or LAN) are added to other Towns as officers, they receive the msg about barbarians and have no idea what Town is being attacked. If it's their own Town or someone else's Town where they have been added.

### LINKED ISSUE #2416: /mc whereami could give better information
### Minecolonies version

version 0.8.8147

### Expected behavior

When using /mc whereami in a colony, it tells you information about the colony (Which it does) And when outside a colony (In the no new colony zone) it says you can't build there, and you're near a colony

### Actual behaviour

When you are outside a colony, but still inside it's no new colony area, /mc whereami will tell the user that they are still in x persons colony, when in reality you aren't

### Steps to reproduce the problem

1. Leave a persons colony, but stay inside the no new colony area
2. Use /mc whereami
3. ...
4. Profit

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2416
Closes #2415
Closes #2391
Closes #2414
Closes #2355

# Changes proposed in this pull request:
- Adds colony where raid happens to message
- Fix whereami message
- Fix issues with not opening inventories
With that, correctly get citizen entities for data.
- Also enable possibility to change miner schematic depending on the style

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 1]
```diff
@@ -108,9 +108,24 @@ public void sendPackets(
...
-                    subscribers.stream()
-                            .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
-                            .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewCitizenViewMessage(colony, citizen), player));
+                    if (citizen.getCitizenEntity().isPresent())
+                    {
+                        final List<EntityCitizen> list = colony.getWorld()
+                                .getEntities(EntityCitizen.class,
+                                        entityCitizen -> entityCitizen.getColony().getID() == colony.getID() && entityCitizen.getCitizenData().getId() == citizen.getId());
+
+                        if (!list.isEmpty() && citizen.getCitizenEntity().get().getEntityId() != list.get(0).getEntityId())
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:117]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 2]
```diff
@@ -29,6 +30,26 @@
  */
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: MAJOR class "EntityAIStructureMiner" has 36 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 75 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 65 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L196](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L196): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L197](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L197): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L198](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L198): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L199](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L199): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L200](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L200): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L835](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L835): Use already-defined constant 'TAG_INVENTORY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 3]
```diff
@@ -29,6 +30,26 @@
...
 public class EntityAIStructureMiner extends AbstractEntityAIStructureWithWorkOrder<JobMiner>
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java:31]: MAJOR class "EntityAIStructureMiner" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 76 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 66 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L196](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L196): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L197](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L197): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L198](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L198): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L199](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L199): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L200](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L200): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L835](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L835): Use already-defined constant 'TAG_INVENTORY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/aad9c6ce324f4e683520045d9707febf734cb756/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**